sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingSymmetricHashJoinExec.scala

```diff
@@ -187,6 +187,12 @@ case class StreamingSymmetricHashJoinExec(
         s"${getClass.getSimpleName} should not take $x as the JoinType")
   }
 
+  override def shouldRunAnotherBatch(newMetadata: OffsetSeqMetadata): Boolean = {
+    (stateWatermarkPredicates.left.nonEmpty || stateWatermarkPredicates.right.nonEmpty) &&
+      eventTimeWatermark.isDefined &&
```

Comment:
nit: we should clearly document that this is the watermark before the current batch